hpx/parallel/execution_policy.hpp

```diff
@@ -65,11 +111,302 @@ namespace hpx { namespace parallel { HPX_INLINE_NAMESPACE(v1)
+    /// sequential_execution_policy.
+    template <typename Executor>
+    struct sequential_task_execution_policy_shim
+      : sequential_task_execution_policy
+    {
+        /// The type of the executor associated with this execution policy
+        typedef Executor executor_type;
+
+        /// The category of the execution agents created by this execution
+        /// policy.
+        typedef typename executor_traits<executor_type>::execution_category
+            execution_category;
+
+        /// Rebind the type of executor used by this execution policy. The
+        /// execution category of Executor shall not be weaker than that of
+        /// this execution policy
+        template <typename Executor_>
+        struct rebind
+        {
+            /// The type of the rebound execution policy
+            typedef sequential_task_execution_policy_shim<Executor_> type;
+        };
+
+        /// \cond NOINTERNAL
+        static std::size_t get_chunk_size() { return 0; }
+        /// \endcond
+
+        /// Create a new sequential_task_execution_policy from itself
+        ///
+        /// \param tag          [in] Specify that the corresponding asynchronous
+        ///                     execution policy should be used
+        ///
+        /// \returns The new sequential_task_execution_policy
+        ///
+        sequential_task_execution_policy_shim operator()(
+            task_execution_policy_tag tag) const
+        {
+            return *this;
+        }
+
+        /// Create a new sequential_task_execution_policy_shim from the given
+        /// executor
+        ///
+        /// \tparam Executor    The type of the executor to associate with this
+        ///                     execution policy.
+        ///
+        /// \param exec         [in] The executor to use for the
+        ///                     execution of the parallel algorithm the
+        ///                     returned execution policy is used with.
+        ///
+        /// \note Requires: is_executor<Executor>::value is true
+        ///
+        /// \returns The new parallel_task_execution_policy
+        ///
+        template <typename Executor_>
+        typename rebind_executor<
+            sequential_task_execution_policy, Executor_
+        >::type
+        on(Executor_& exec) const
+        {
+            BOOST_STATIC_ASSERT(is_executor<Executor_>::value);
+
+            typedef typename rebind_executor<
+                sequential_task_execution_policy, Executor_
+            >::type rebound_type;
+            return rebound_type(exec);
+        }
+
+        /// Return the associated executor object.
+        Executor& executor() { return exec_; }
+        /// Return the associated executor object.
+        Executor const& executor() const { return exec_; }
+
+    private:
+        /// \cond NOINTERNAL
+        friend struct sequential_task_execution_policy;
+        template <typename> friend struct sequential_execution_policy_shim;
```

Comment:
Should this be `template <typename> friend struct sequential_task_execution_policy_shim` since `sequential_execution_policy_shim::rebind::type` typedefs from a `sequential_execution_policy_shim`?